test/distributed/_tensor/test_device_mesh.py

```diff
@@ -16,13 +19,32 @@
     DTensorTestBase,
     with_comms,
 )
+from torch.testing._internal.common_distributed import TEST_SKIPS
 
 
 class DeviceMeshTest(DTensorTestBase):
     @property
     def world_size(self):
         return 8
 
+    def test_init_process_group(self):
+        self.device_type = "cuda" if torch.cuda.is_available() else "cpu"
+        backend = "nccl" if self.device_type == "cuda" else "gloo"
+        # skip the test if not enough GPUs
+        if backend == "nccl" and torch.cuda.device_count() < self.world_size:
+            sys.exit(TEST_SKIPS[f"multi-gpu-{self.world_size}"].exit_code)
+        mesh_tensor = torch.arange(4).reshape(2, 2)
+        self.assertTrue(not is_initialized())
+        os.environ["MASTER_ADDR"] = "localhost"
+        os.environ["MASTER_PORT"] = "25364"
+        # mesh size can be smaller than world_size
```

Comment:
discussed offline, just do some simple assertion to ensure the num of ranks in mesh is the same as the world_size initially

Reply:
#91801

Reply:
nit: remove this comment?